lc/Cargo.toml

```diff
@@ -1,6 +1,6 @@
 [package]
 edition = "2021"
-version = "0.2.0"
+version = "0.3.0"
```

Comment:
This to me looks like a refactor. I would go to 0.2.1

Reply:
I based it on [this](https://semver.org/):
> MINOR version when you add functionality in a backward compatible manner

rather than
> PATCH version when you make backward compatible bug fixes

because these commits do not fix any bugs!

Reply:
There is a bit of debate about this (https://github.com/semver/semver/issues/146)
To me it is fine to go on with `minor` for refactors. 
In any case I would reword the first 4 commits as refactors.